aiohttp/web_urldispatcher.py

```diff
@@ -104,23 +116,23 @@ class _InfoDict(TypedDict, total=False):
 
     directory: Path
     prefix: str
-    routes: Mapping[str, "AbstractRoute"]
+    routes: Mapping[str, AbstractRoute]
 
-    app: "Application"
+    app: Application
 
     domain: str
 
-    rule: "AbstractRuleMatching"
+    rule: AbstractRuleMatching
 
     http_exception: HTTPException
 
 
 class AbstractResource(Sized, Iterable["AbstractRoute"]):
-    def __init__(self, *, name: Optional[str] = None) -> None:
+    def __init__(self, *, name: str | None = None) -> None:
```

Comment:
We've been avoiding these typing changes. This is not valid on Python 3.9 (future import only delays the error, it will still happen if you inspect the types at runtime).

Pyupgrade will automatically update all of these across the entire project once 3.9 is dropped.

Reply:
But aiohttp never inspects the types at runtime!
How can it can affect anybody?

Reply:
A user could inspect them and (e.g.) pass the signature into pydantic or something.

Reply:
Ok, I'll revert because typing changes not a subject of the PR.
I still think that type hints from aiohttp are used by type checkers only and never in runtime.

Reply:
> I still think that type hints from aiohttp are used by type checkers only and never in runtime.

Well, that's down to the user. We've seen much weirder things. As I say, pyupgrade will handle all of this automatically anyway.